src/common/src/main/java/org/niis/xrd4j/common/util/SOAPHelper.java

```diff
@@ -587,16 +598,23 @@ public static void updateNamespaceAndPrefix(NodeList list, String namespace, Str
      * @param namespace target namespace
      * @param prefix    target prefix
      * @return updated Node
+     * @throws SOAPException if renaming xml node throws DOMException
      */
-    public static Node updateNamespaceAndPrefix(Node node, String namespace, String prefix) {
-        if (node.getNodeType() == ELEMENT_NODE) {
+    public static Node updateNamespaceAndPrefix(Node node, String namespace, String prefix) throws SOAPException {
+        try {
+            if (!(node.getNodeType() == ELEMENT_NODE)) {
+                return node;
+            }
+            ElementImpl elementImpl = (ElementImpl) node;
```

Comment:
Just want to note that this might throw a `RuntimeException`, specifically the [ClassCastException](https://docs.oracle.com/en/java/javase//21/docs/api/java.base/java/lang/ClassCastException.html) if the node can't actually be cast to the specific implementation. But we likely don't have a good way to resolve this, as even if we were to wrap this in another exception, there isn't much the application could do to recover from this case.